### DISCUSSION TITLE
Is there an `entered` action?

### DISCUSSION THREAD

**Discussion by gonnavis** [statelyai/xstate#2734] [Q&A]
`entry` action fires before transitioning to this state.

Is there an `entered` action?

**Comment by Andarist** [comment]
When do you observe a difference? It would be nice to get some more context behind this question. Is it about `assign` action receiving a `{ state }` before a transition in its third argument?
```js
assign((ctx, ev, { state }) => ({}))
```

**Comment by gonnavis** [comment]
Hello, I have a custom entry function.
In it, I use service.state.matches() to check the current state, ~~but it matches the previous state, not the current~~.

Oh, because `onTransition` event fires after `entry` action, so I guess it will not match, but I tried again found that it does match.
No problem now. Thanks!

**Reply by Andarist** [reply]
q: if you want to know the current state to execute some action conditionally~ then why don't you put `entry` in the desired state? By implementing the conditional logic within some other `entry` or within `onTransition` you make this logic implicitly related to the machine. By making it part of the machine you would end up with a more understandable model and it could be part of the visualization and all
